src/Features/Core/Portable/ExtractClass/AbstractExtractClassRefactoringProvider.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable 
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.PullMemberUp;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.ExtractClass
+{
+    internal abstract class AbstractExtractClassRefactoringProvider : CodeRefactoringProvider
+    {
+        private readonly IExtractClassOptionsService? _optionsService;
+
+        public AbstractExtractClassRefactoringProvider() : this(service: null)
+        {
+        }
+
+        public AbstractExtractClassRefactoringProvider(IExtractClassOptionsService? service)
+        {
+            _optionsService = service;
+        }
+
+        protected abstract Task<SyntaxNode?> GetSelectedNodeAsync(CodeRefactoringContext context);
```

Comment:
i think this should be doing whatever our 'select members' features are doing.  i.e. it's totally reasonable to select N members and want to 'pull up to new base class'.

Reply:
I'll add that for this and Pull Members Up in a follow up PR. Filed #46438 to track